Controllers/Frontend/BaseFrontendController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+namespace MakairaConnect\Controllers\Frontend;
+
+use Enlight_Controller_Action;
+use Enlight_Controller_Exception;
+use Enlight_Controller_Response_ResponseHttp;
+use Exception;
+use Makaira\Signing\Hash\Sha256;
+use Shopware\Components\CSRFWhitelistAware;
+use Shopware\Components\Plugin\Configuration\ReaderInterface;
+
+class BaseFrontendController extends Enlight_Controller_Action implements CSRFWhitelistAware
+{
+    /**
+     * Pre dispatch method
+     * @throws Enlight_Controller_Exception
+     */
+    public function preDispatch()
+    {
+        $this->container->get('plugin_manager')->Controller()->ViewRenderer()->setNoRender();
+        $configReader = $this->container->get(ReaderInterface::class);
+        $config = $configReader->getByPluginName('MakairaConnect');
+
+        $result = $this->verifySignature($config['makaira_connect_secret']);
+        if ($result instanceof Enlight_Controller_Response_ResponseHttp) {
+            $result->send();
+            exit;
+        }
+    }
+
+    /**
+     * @param string $secret
+     * @return ?Enlight_Controller_Response_ResponseHttp
+     */
+    private function verifySignature(string $secret): ?Enlight_Controller_Response_ResponseHttp
+    {
+        if (!$this->request->headers->has('x-makaira-nonce') ||
+            !$this->request->headers->has('x-makaira-hash')) {
+            return $this->createResponse([
+                'message' => 'Unauthorized'
+            ], 401);
+        }
+
+        $signer = new Sha256();
+
+        $expected = $signer->hash(
+            $this->request->headers->get('x-makaira-nonce'),
+            $this->request->getContent(),
+            $secret
+        );
+
+        $current = $this->request->headers->get('x-makaira-hash');
+
+        if (!hash_equals($expected, $current)) {
+            return $this->createResponse([
+                'message' => 'Forbidden'
+            ], 403);
+        }
+
+        return null;
+    }
```

Comment:
The endpoints will be queried by the storefront and it can't send signed requests. Reasons for it:
- The `secret` will be exposed to the browser and therefor to other JS scripts and browser extensions.
- An external library is required to create the signature.

```suggestion
```